packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -45,21 +61,36 @@ export class PropertiesManager {
         newProps: PropertySet,
         op?: ICombiningOp,
         seq?: number,
-        collaborating: boolean = false): PropertySet | undefined {
+        collaborating: boolean = false,
+        rollback: PropertiesRollback = PropertiesRollback.None): PropertySet | undefined {
         if (!this.pendingKeyUpdateCount) {
             this.pendingKeyUpdateCount = createMap<number>();
         }
 
         // There are outstanding local rewrites, so block all non-local changes
-        if (this.pendingRewriteCount > 0 && seq !== UnassignedSequenceNumber && collaborating) {
+        if (this.pendingRewriteCount > 0 && seq !== UnassignedSequenceNumber && seq !== UniversalSequenceNumber
+            && collaborating) {
             return undefined;
         }
 
+        // Clean up counts for rolled back edits before modifying oldProps
+        if (collaborating) {
+            if (rollback === PropertiesRollback.Rollback) {
+                this.decrementPendingCounts(false, newProps);
+            } else if (rollback === PropertiesRollback.Rewrite) {
+                // oldProps is the correct props for tracking counts on rewrite because the ones in newProps include
+                // those that were implicitly cleared by the rewrite for which we don't track pending counts.
+                // This doesn't handle the edge case for rewrite with a prop set to null to remove it redundantly.
```

Comment:
Any opinions on whether it's worthwhile to fix this? If so, rather than propagate more op info for rollbacks, I'd prevent tracking such a property change increment in the first place.

Reply:
seems simple enough to fix on L134-144, right? I think we might as well if it's not involved. Up to you though, overall this look good to me.

Reply:
Another reason to fix this: the delta is wrong for this case even without rollback.